device/device.go

```diff
@@ -0,0 +1,110 @@
+package device
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/josephroberts/edge-node-manager/device/deviceType"
+	"github.com/josephroberts/edge-node-manager/radio/bluetooth"
+	"github.com/josephroberts/edge-node-manager/radio/radioType"
+	"github.com/josephroberts/edge-node-manager/radio/wifi"
+	"github.com/josephroberts/edge-node-manager/radio/zigbee"
+)
+
+type Type struct {
+	Device deviceType.Type
+	Radio  radioType.Type
+}
+
+func (t Type) Scan(name string, timeout time.Duration) ([]string, error) {
+	switch t.Radio {
+	case radioType.BLUETOOTH:
+		return bluetooth.Scan(name, timeout)
+	case radioType.WIFI:
+		return wifi.Scan(name, timeout)
+	case radioType.ZIGBEE:
+		return zigbee.Scan(name, timeout)
+	default:
+		return nil, fmt.Errorf("Radio %s does not exist", t.Radio)
+	}
+}
+
+func Create(t *Type) Interface {
+	switch t.Device {
+	case deviceType.NRF51822:
+		return &Nrf51822{
+			Device: &Device{
+				Type: t,
+			},
+		}
+	case deviceType.ESP8266:
+		return &Esp8266{
+			Device: &Device{
+				Type: t,
+			},
+		}
+	}
+
+	return nil
+}
+
+func Provision(t *Type, localUUID, applicationUUID, resinUUID string) Interface {
+	device := Create(t)
+	device.GetDevice().LocalUUID = localUUID
```

Comment:
Here you really should just assign a var to `.GetDevice()`.

Also this is so horribly confusing here. `device := X` `device.GetDevice().foo = bar` etc. etc. yikes.

I'd rename accordingly and maybe add a comment about the whole interface dance, if you are unable to rearch away from it.